src/test/java/julia/git/CommitTest.java

```diff
@@ -48,47 +48,48 @@ public void possibleToGetHashOfCommit() {
-        commit2.setParent(commit3.getHash());
-        assertNotEquals("3c6b4a6afd7ec87955f6b4ccb3cec6b774b782bf", commit2.getHash());
+        Commit commit2 = new Commit(directory.getHash(), commit.getHash(), "Test", "gorshkova_julia", "Tue Apr 27 22:32:35 2021 +0300");
+        assertNotEquals(commit.getHash(), commit2.getHash());
     }
 
     @Test
     public void hashOfCommitIsChangedWhenMessageIsChanged() {
-        byte[] b = new byte[] {116, 101, 115, 116};
+        byte[] b = new byte[]{116, 101, 115, 116};
         File file = new File(b);
         Directory directory = new Directory();
         directory.add("Tester", file);
         Commit commit = new Commit(directory.getHash(), "Test", "gorshkova_julia", "Tue Apr 27 22:32:35 2021 +0300");
-        commit.setMessage("Testik");
-        assertNotEquals("932af51418307f9edf6387a183cebc8735c34c3e", commit.getHash());
+        Commit commit2 = new Commit(directory.getHash(), commit.getHash(), "Test", "gorshkova_julia", "Tue Apr 27 22:32:35 2021 +0300");
+        Commit commit3 = new Commit(directory.getHash(), commit.getHash(),"Testik", "gorshkova_julia", "Tue Apr 27 22:32:35 2021 +0300");
+        assertNotEquals(commit2.getHash(), commit3.getHash());
     }
 
     @Test
     public void hashOfCommitIsChangedWhenAuthorIsChanged() {
-        byte[] b = new byte[] {116, 101, 115, 116};
+        byte[] b = new byte[]{116, 101, 115, 116};
         File file = new File(b);
         Directory directory = new Directory();
         directory.add("Tester", file);
         Commit commit = new Commit(directory.getHash(), "Test", "gorshkova_julia", "Tue Apr 27 22:32:35 2021 +0300");
-        commit.setAuthor("ivanov_ivan");
-        assertNotEquals("932af51418307f9edf6387a183cebc8735c34c3e", commit.getHash());
+        Commit commit2 = new Commit(directory.getHash(), commit.getHash(),"Test", "gorshkova_julia", "Tue Apr 27 22:32:35 2021 +0300");
+        Commit commit3 = new Commit(directory.getHash(), commit.getHash(),"Test", "ivanov_ivan", "Tue Apr 27 22:32:35 2021 +0300");
+        assertNotEquals(commit2.getHash(), commit3.getHash());
     }
 
     @Test
     public void hashOfCommitIsChangedWhenDateIsChanged() {
-        byte[] b = new byte[] {116, 101, 115, 116};
+        byte[] b = new byte[]{116, 101, 115, 116};
         File file = new File(b);
         Directory directory = new Directory();
         directory.add("Tester", file);
         Commit commit = new Commit(directory.getHash(), "Test", "gorshkova_julia", "Tue Apr 27 22:32:35 2021 +0300");
-        commit.setDate("Tue Jun 19 22:32:35 2021 +0300");
-        assertNotEquals("932af51418307f9edf6387a183cebc8735c34c3e", commit.getHash());
+        Commit commit2 = new Commit(directory.getHash(), commit.getHash(), "Test", "gorshkova_julia", "Tue Apr 27 22:32:35 2021 +0300");
+        Commit commit3 = new Commit(directory.getHash(), commit.getHash(), "Test", "gorshkova_julia", "Tue Apr 28 22:32:35 2021 +0300");
```

Comment:
BTW, if these are going to be popular operations (creating a new commit based on another), we could create methods like this:

```
class Commit {
    ...
    Commit withDate(String newDate) {
         return new Commit(this.message, ..., newDate);
    }
    ...
}
```